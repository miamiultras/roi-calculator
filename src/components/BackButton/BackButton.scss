.back-btn {
  margin-top: 25px;
  a {
    text-decoration: none;
    &:visited {
      text-decoration: none;
    }
  }
  .container {
    display: flex;
    justify-content: center;
    
    button {
      width: 100%;
      display: flex;
      justify-content: space-around;
      align-items: center;
      font-family: 'Roboto';
      font-size: 16px;
      background: transparent;
      color: #5db98f;
      cursor: pointer;
      text-transform: uppercase;
      width: auto;
      padding: 10px;
      border: none;

      i {
        border: solid #5db98f;
        border-width: 0 1.2px 1.2px 0;
        padding: 3px;
        margin-right: 3px;
      } 
    
      .left {
        transform: rotate(135deg);
        -webkit-transform: rotate(135deg);
      }
    
    }
  }
}