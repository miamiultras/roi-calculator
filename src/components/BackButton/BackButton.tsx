import React from 'react';
import { Link } from 'react-router-dom';

import './BackButton.scss';

const BackButton = ({ name, to }: { name: string, to: string }) => (
  <div className='back-btn'>
    <Link to={`/calculator/${to}`}>
      <div className='container'>
        <button>
          <i className='arrow left' />
          <span>
            {name}
          </span>
        </button>
      </div>
    </Link>
  </div>
);

export default BackButton;
