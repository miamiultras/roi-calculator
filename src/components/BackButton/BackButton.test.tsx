import * as React from 'react';
import { shallow } from 'enzyme';
import BackButton from './BackButton';

function setup() {
  const props = {};
  const wrapper = shallow(<BackButton name='test' to='test' />);
  return {
    wrapper,
    props,
  };
}

describe('BackButton component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
