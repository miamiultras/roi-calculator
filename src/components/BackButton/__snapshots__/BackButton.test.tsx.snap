// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackButton
    name="test"
    to="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        replace={false}
        to="/calculator/test"
      >
        <div
          className="container"
        >
          <button>
            <i
              className="arrow left"
            />
            <span>
              test
            </span>
          </button>
        </div>
      </Link>,
      "className": "back-btn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
        >
          <button>
            <i
              className="arrow left"
            />
            <span>
              test
            </span>
          </button>
        </div>,
        "replace": false,
        "to": "/calculator/test",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button>
            <i
              className="arrow left"
            />
            <span>
              test
            </span>
          </button>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="arrow left"
              />,
              <span>
                test
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "arrow left",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test",
              },
              "ref": null,
              "rendered": "test",
              "type": "span",
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          replace={false}
          to="/calculator/test"
        >
          <div
            className="container"
          >
            <button>
              <i
                className="arrow left"
              />
              <span>
                test
              </span>
            </button>
          </div>
        </Link>,
        "className": "back-btn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="container"
          >
            <button>
              <i
                className="arrow left"
              />
              <span>
                test
              </span>
            </button>
          </div>,
          "replace": false,
          "to": "/calculator/test",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              <i
                className="arrow left"
              />
              <span>
                test
              </span>
            </button>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="arrow left"
                />,
                <span>
                  test
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow left",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test",
                },
                "ref": null,
                "rendered": "test",
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
